src/main/java/nl/novi/catsittermanager/controllers/CatController.java

```diff
@@ -40,7 +40,11 @@ public ResponseEntity<List<CatDto>> getAllCats() {
     @GetMapping("/{id}")
     public ResponseEntity<CatDto> getCat(@PathVariable("id") final UUID idToFind) {
         CatDto catDto = catService.getCat(idToFind);
-        return ResponseEntity.ok(catDto);
+        if (catDto == null) {
+            return ResponseEntity.notFound().build();
+        } else {
+            return ResponseEntity.ok(catDto);
+        }
     }
 
     @PostMapping
```

Comment:
same as  above., else part is not needed. Move everything from line 55 to 60 outside the else part